src/Facets/EcoFacet.sol

```diff
@@ -0,0 +1,256 @@
+
+        _startBridge(_bridgeData, _ecoData);
+    }
+
+    /// @notice Swaps and bridges tokens via Eco Protocol
+    /// @param _bridgeData Bridge data containing core parameters
+    /// @param _swapData Array of swap data for source swaps
+    /// @param _ecoData Eco-specific parameters for the bridge
+    function swapAndStartBridgeTokensViaEco(
+        ILiFi.BridgeData memory _bridgeData,
+        LibSwap.SwapData[] calldata _swapData,
+        EcoData calldata _ecoData
+    )
+        external
+        payable
+        nonReentrant
+        refundExcessNative(payable(msg.sender))
+        containsSourceSwaps(_bridgeData)
+        validateBridgeData(_bridgeData)
+    {
+        // Validate eco-specific data before swapping
+        _validateEcoData(_bridgeData, _ecoData);
+
+        // For ERC20 tokens, we need to reserve the solver reward from the swapped amount
+        // Only pass native fee reservation if the final asset is native
+        uint256 nativeFeeAmount = LibAsset.isNativeAsset(
+            _bridgeData.sendingAssetId
+        )
+            ? _ecoData.solverReward
+            : 0;
+
+        _bridgeData.minAmount = _depositAndSwap(
+            _bridgeData.transactionId,
+            _bridgeData.minAmount,
+            _swapData,
+            payable(msg.sender),
+            nativeFeeAmount
+        );
+
+        // For ERC20 tokens, the swap result includes the solver reward
+        // We need to subtract it to get the actual bridge amount
+        if (!LibAsset.isNativeAsset(_bridgeData.sendingAssetId)) {
+            _bridgeData.minAmount =
+                _bridgeData.minAmount -
+                _ecoData.solverReward;
+        }
+
+        _startBridge(_bridgeData, _ecoData);
+    }
+
+    /// Internal Methods ///
+
+
+    function _getEcoChainId(
+        uint256 _lifiChainId
+    ) private pure returns (uint64) {
+        // Map LiFi Tron chain ID to Eco protocol Tron chain ID
+        if (_lifiChainId == LIFI_CHAIN_ID_TRON) {
+            return 728126428; // Eco protocol's Tron chain ID
+        }
+        // Map LiFi Solana chain ID to Eco protocol Solana chain ID
+        if (_lifiChainId == LIFI_CHAIN_ID_SOLANA) {
+            return 1399811149; // Eco protocol's Solana chain ID
+        }
+        // For all other chains (EVM), pass through as-is
+        return uint64(_lifiChainId);
+    }
+
+    function _buildReward(
+        ILiFi.BridgeData memory _bridgeData,
+        EcoData calldata _ecoData
+    ) private view returns (IEcoPortal.Reward memory) {
+        bool isNative = LibAsset.isNativeAsset(_bridgeData.sendingAssetId);
+
+        IEcoPortal.TokenAmount[] memory rewardTokens;
+        if (!isNative) {
+            rewardTokens = new IEcoPortal.TokenAmount[](1);
+            rewardTokens[0] = IEcoPortal.TokenAmount({
+                token: _bridgeData.sendingAssetId,
+                amount: _bridgeData.minAmount + _ecoData.solverReward
```

Comment:
_ecoData.solverReward is described above as NATIVE token amount. 
Here you are using it as an ERC20 token amount. Is this really what you want to do here?